rice-framework/krad-web-framework/src/main/java/org/kuali/rice/krad/document/DocumentControllerServiceImpl.java

```diff
@@ -812,17 +813,17 @@ public void performWorkflowAction(DocumentFormBase form, UifConstants.WorkflowAc
                     successMessageKey = RiceKeyConstants.MESSAGE_SAVED;
                     break;
                 case ROUTE:
-                    document = getDocumentService().routeDocument(document, form.getAnnotation(),
+                    document = getDocumentService().routeDocument(document, form.getAnnotation(), nodeName,

```

Comment:
As with KualiDocumentActionBase below, I don't think you need to do this, just call the version of routedocument without passing the node name, it's going to do the same thing you've done manually above by invoking document.getAdHocRouteNodeName

This applies in the other parts of this file as well, I don't think you even need to change this class. That keeps it a little bit simpler and cleaner in my mind. But at the end of the day the behavior's the same.